webscraping/webscrape.py

```diff
@@ -0,0 +1,47 @@
+import http.client
+
+def firstBetween (string, startingString, endingString): #This function return whatever is between the starting string and ending string based off of an original string
+    idx = [i for i in range(len(string)) if string.startswith(startingString, i)]
+    if len(idx) > 0:
+        ret = exerciseHTML[idx[0]+ len(startingString):].split(endingString)[0]
+        return ret
+    return False
+
+def requestHTML(reqName):
+    conn = http.client.HTTPSConnection("www.acefitness.org")
+    conn.request("GET", reqName)
+    r1 = conn.getresponse()
+    print(r1.status, r1.reason)
+    data1 = r1.read()  # This will return entire content.
+    # The following example demonstrates reading data in chunks.
+    conn.request("GET",  reqName)
+    r1 = conn.getresponse()
+    OUTPUT = ""
+    with open('file.txt', 'w') as file:
+        file.write("")
+    while chunk := r1.read(200):
+        with open('file.txt', 'a') as file:
+            file.write(repr(chunk)[2:-1])
+            file.close()
+    with open('file.txt', 'r') as file:
+        OUTPUT = file.read()
+        formatted_output = OUTPUT.replace('\\n', '\n').replace('\\t', '\t').replace('\\r', '')
+        file.close()
+    with open('file.txt', 'w') as file:
+        file.write(formatted_output)
+        file.close()
+    return OUTPUT
+
+for idx in range(30):
```

Comment:
Maybe extract this logic into a separate function processLinks()

Reply:
Good idea for modularization